uaclient/api/u/pro/services/enable/v1.py

```diff
@@ -0,0 +1,170 @@
+import logging
+from typing import List, Optional
+
+from uaclient import entitlements, event_logger, lock, messages, util
+from uaclient.api import exceptions
+from uaclient.api.api import APIEndpoint
+from uaclient.api.data_types import AdditionalInfo
+from uaclient.api.u.pro.status.enabled_services.v1 import _enabled_services
+from uaclient.api.u.pro.status.is_attached.v1 import _is_attached
+from uaclient.config import UAConfig
+from uaclient.data_types import (
+    BoolDataValue,
+    DataObject,
+    Field,
+    StringDataValue,
+    data_list,
+)
+
+event = event_logger.get_event_logger()
+LOG = logging.getLogger(util.replace_top_level_logger_name(__name__))
+
+
+class EnableOptions(DataObject):
+    fields = [
+        Field("service", StringDataValue),
+        Field("variant", StringDataValue, False),
+        Field("enable_required_services", BoolDataValue, False),
+        Field("disable_incompatible_services", BoolDataValue, False),
+        Field("access_only", BoolDataValue, False),
+    ]
+
+    def __init__(
+        self,
+        *,
+        service: str,
+        variant: Optional[str] = None,
+        enable_required_services: bool = True,
+        disable_incompatible_services: bool = True,
+        access_only: bool = False
+    ):
+        self.service = service
+        self.variant = variant
+        self.enable_required_services = enable_required_services
+        self.disable_incompatible_services = disable_incompatible_services
+        self.access_only = access_only
+
+
+class EnableResult(DataObject, AdditionalInfo):
+    fields = [
+        Field("enabled", data_list(StringDataValue)),
+        Field("disabled", data_list(StringDataValue)),
+        Field("reboot_required", BoolDataValue),
+        Field("messages", data_list(StringDataValue)),
+    ]
+
+    def __init__(
+        self,
+        *,
+        enabled: List[str],
+        disabled: List[str],
+        reboot_required: bool,
+        messages: List[str]
+    ):
+        self.enabled = enabled
+        self.disabled = disabled
+        self.reboot_required = reboot_required
+        self.messages = messages
+
+
+def _enabled_services_names(cfg: UAConfig) -> List[str]:
+    return [s.name for s in _enabled_services(cfg).enabled_services]
+
+
+def enable(options: EnableOptions) -> EnableResult:
+    return _enable(options, UAConfig())
+
+
+def _enable(
+    options: EnableOptions,
+    cfg: UAConfig,
+) -> EnableResult:
+    event.set_event_mode(event_logger.EventLoggerMode.JSON)
+
+    if not _is_attached(cfg).is_attached:
```

Comment:
Should we also handle the situation where a user is running as `non-root` ?

Reply:
added!